app/models/concerns/foreman_azure_rm/vm_extensions/managed_vm.rb

```diff
@@ -33,10 +32,40 @@ def define_managed_storage_profile(vm_name, os_disk_caching, platform, premium_o
                                                    end
         os_disk.managed_disk = managed_disk_params
         storage_profile.os_disk = os_disk
-
         storage_profile
       end
 
+      def define_data_disks(vm_name, data_disks)
+        unless data_disks.nil?
+          disks = []
+          disk_count = 0
+          data_disks.each do |disk_num, attrs|
+            managed_data_disk = ComputeModels::ManagedDiskParameters.new
+            disk = ComputeModels::DataDisk.new
+            disk.name = "#{vm_name}-data-disk#{disk_count}"
+            disk.caching = if attrs[:data_disk_caching].present?
```

Comment:
We can remove the `if` statement and leave only the `case` with extra `when nil` clause.